packages/react-server/src/ReactFlightServer.js

```diff
@@ -364,6 +364,17 @@ function describeValueForErrorMessage(value: ReactModel): string {
     }
     case 'function':
       return 'function';
+    case 'undefined':
+      let info = 'undefined';
+
+      if (__DEV__) {
+        info +=
+          '. You likely forgot to export your component from the file ' +
+          "it's defined in, or you might have mixed up default and " +
+          'named imports.';
+      }
+
+      return info;
```

Comment:
`describeValueForErrorMessage` is called from many places, not just when we have an invalid component type so this explanatory text may be misleading in some cases.

I think the intent to clarify here is good but I'd maybe add the logic to line 243 and not what is essentially a custom `toString` for error messages in this file